### PR TITLE
License locations

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Store the locations of license findings.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1158)

### CODE REVIEW

**Comment by tsteenbe** [comment]
Looking it at the code should copyrights also have a location with startLine and endLine so user can easily find the match (similar to why we include lines with licenses) would also be possible to suppress copyrightholder false-positive

File: model/src/test/kotlin/ScanResultContainerTest.kt [hunk 1]
```diff
@@ -22,6 +22,7 @@ package com.here.ort.model
 import com.fasterxml.jackson.module.kotlin.readValue
 
 import com.here.ort.utils.test.patchActualResult
+import io.kotlintest.matchers.sorted
```
[the remark this answers is not in the document]
Comment (mnonnenmacher) [reply model/src/test/kotlin/ScanResultContainerTest.kt]: Done.

File: utils/src/main/kotlin/SortedSetComparator.kt [hunk 2]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2017-2018 HERE Europe B.V.
```
Comment (sschuberth) [inline utils/src/main/kotlin/SortedSetComparator.kt]: Nit: This should be `-2019` by now.

File: model/src/main/kotlin/TextLocation.kt [hunk 3]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2017-2018 HERE Europe B.V.
```
Comment (sschuberth) [inline model/src/main/kotlin/TextLocation.kt]: Nit: This should be `-2019` by now.

File: model/src/main/kotlin/LicenseFinding.kt [hunk 4]
```diff
@@ -53,18 +53,23 @@ fun LicenseFindingsMap.removeGarbage(copyrightGarbage: CopyrightGarbage) =
  */
 data class LicenseFinding @JsonCreator constructor(
         val license: String,
+        val locations: SortedSet<TextLocation> = sortedSetOf(),
```
Comment (sschuberth) [inline model/src/main/kotlin/LicenseFinding.kt]: I'm wonder why we should assign a default value to this, but not to `copyrights`. Sure, not all scanners might provide `locations`, but the same is true for `copyrights`. Would it require any additional code changes currently to simply omit the default value here? Or is it simply for backward compatibility for deserializing older results?

File: utils/src/main/kotlin/SortedSetComparator.kt (hunk 2 above)
Comment (mnonnenmacher) [reply utils/src/main/kotlin/SortedSetComparator.kt]: Done.

File: model/src/main/kotlin/TextLocation.kt (hunk 3 above)
Comment (mnonnenmacher) [reply model/src/main/kotlin/TextLocation.kt]: Done.

File: model/src/main/kotlin/LicenseFinding.kt (hunk 4 above)
Comment (mnonnenmacher) [reply model/src/main/kotlin/LicenseFinding.kt]: It's for backward compatibility. It's the easiest way to ensure old results can be deserialized.

Comment (sschuberth) [reply model/src/main/kotlin/LicenseFinding.kt]: Yes, I'd have a slight preference for that for clarity in this case.

Comment (mnonnenmacher) [reply model/src/main/kotlin/LicenseFinding.kt]: Done.

**Comment by sschuberth** [comment]
@mnonnenmacher CI fails due to a compiler warning, it seems.

File: model/src/main/kotlin/LicenseFinding.kt (hunk 4 above)
Comment (mnonnenmacher) [reply model/src/main/kotlin/LicenseFinding.kt]: I had to replace the `@JsonCreator` constructors with a custom deserializer, see the commit message for details.

**Pull request merged**